File: software/examples/javascript/ExampleScanForTags.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your NFC Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var nfc = new Tinkerforge.BrickletNFC(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Enable reader mode
        nfc.setMode(Tinkerforge.BrickletNFC.MODE_READER);
    }
);

// Register reader state changed callback
nfc.on(Tinkerforge.BrickletNFC.CALLBACK_READER_STATE_CHANGED,
    // Callback function for reader state changed callback
    function (state, idle) {
        if(state == Tinkerforge.BrickletNFC.READER_STATE_REQUEST_TAG_ID_READY) {
            nfc.readerGetTagID(
                function (tagType, tagID) {
                    var tagInfo = ''

                    for (var i = 0; i < tagID.length; i++) {
                        tagInfo += '0x' + ('0' + tagID[i].toString(16).toUpperCase()).substr(-2)

                        if (i < tagID.length - 1) {
                            tagInfo += ' '
                        }
                    }

                    console.log('Found tag of type %d with ID [%s]', tagType, tagInfo)
                },
                function (error) {
                    console.log('Could not get tag ID: ' + error)
                }
            );
        }

        if(idle) {
            nfc.readerRequestTagID();
        }
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/csharp/ExampleScanForTags.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your NFC Bricklet

	// Callback function for reader state changed callback
	static void ReaderStateChangedCB(BrickletNFC sender, byte state, bool idle)
	{
		if(state == BrickletNFC.READER_STATE_REQUEST_TAG_ID_READY)
		{
			byte tagType;
			byte[] tagID;
			String tagInfo;

			sender.ReaderGetTagID(out tagType, out tagID);

			tagInfo = String.Format("Found tag of type {0} with ID [", tagType);

			for (int i = 0; i < tagID.Length; i++)
			{
				tagInfo += String.Format("0x{0:02X}", tagID[i]);

				if (i < tagID.Length - 1)
				{
					tagInfo += " ";
				}
			}

			tagInfo += "]";

			Console.WriteLine(tagInfo);
		}

		if(idle)
		{
			sender.ReaderRequestTagID();
		}
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletNFC nfc = new BrickletNFC(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register reader state changed callback to function ReaderStateChangedCB
		nfc.ReaderStateChangedCallback += ReaderStateChangedCB;

		// Enable reader mode
		nfc.SetMode(BrickletNFC.MODE_READER);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}
